src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/IndustrialMiner.java

```diff
@@ -0,0 +1,213 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items.multiblocks.miner;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import io.github.thebusybiscuit.cscorelib2.chat.ChatColors;
+import io.github.thebusybiscuit.cscorelib2.item.CustomItem;
+import io.github.thebusybiscuit.slimefun4.api.MinecraftVersion;
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineFuel;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.multiblocks.MultiBlockMachine;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+
+/**
+ * The {@link IndustrialMiner} is a {@link MultiBlockMachine} that can mine any
+ * ores it finds in a given range underneath where it was placed.
+ * 
+ * <i>And for those of you who are wondering... yes this is the replacement for the
+ * long-time deprecated Digital Miner.</i>
+ * 
+ * @author TheBusyBiscuit
+ * 
+ * @see AdvancedIndustrialMiner
+ * @see ActiveMiner
+ *
+ */
+public class IndustrialMiner extends MultiBlockMachine {
+
+    protected final Map<Location, ActiveMiner> activeMiners = new HashMap<>();
+    protected final List<MachineFuel> fuelTypes = new ArrayList<>();
```

Comment:
does this need to be a list? If you need order a LinkedHashSet may be better.

Reply:
It does need to be consistently ordered, don't really see the benefits of using a LinkedHashSet though here, an ArrayList is just fine for this purpose.